internal/provider/kubernetes/predicates.go

```diff
@@ -294,7 +294,7 @@ func (r *gatewayAPIReconciler) validateServiceForReconcile(obj client.Object) bo
 	// Check if the Service belongs to a Gateway, if so, update the Gateway status.
 	gtw := r.findOwningGateway(ctx, labels)
 	if gtw != nil {
-		r.updateStatusForGateway(ctx, gtw)
+		r.resources.GatewayStatuses.Store(utils.NamespacedName(gtw), &gtw.Status)
```

Comment:
this is not documented well, but the reason this existed, was so that this flow could update the `Status.Address` field and `Status.Conditions` while the other flow (published from Gateway API) would update `Status.Conditions`

wanted to highlight this

Reply:
Yep, better to add a line in the comment to explain why need to send an update here and how the gateway address is updated.